File: CORREDOR/4 Lenguajes/Go/Metodos_SinParametros.go
package main

import "fmt"

// ----- Estructura MetodosSinParametros -----
type MetodosSinParametros struct{}

func (m MetodosSinParametros) Arrancar() {
	fmt.Println("El coche ha arrancado.")
}

func (m MetodosSinParametros) Detener() {
	fmt.Println("El coche se ha detenido.")
}

type Persona struct{}

func (p Persona) Saludar() {
	fmt.Println("Hola, mucho gusto!")
}

func (p Persona) Dormir() {
	fmt.Println("La persona está durmiendo...")
}

type Perro struct{}

func (pe Perro) Comer() {
	fmt.Println("El perro está comiendo su comida.")
}

func (pe Perro) Ladrar() {
	fmt.Println("El perro está ladrando ¡Guau guau!")
}

func main() {
	mp := MetodosSinParametros{}
	mp.Arrancar()
	mp.Detener()

	p := Persona{}
	p.Saludar()
	p.Dormir()

	pe := Perro{}
	pe.Comer()
	pe.Ladrar()
}

File: CORREDOR/4 Lenguajes/C#/ProyectoCoche/Metodos_SinParametros.cs
using System;

namespace Java
{
    class Metodos_SinParametros
    {
        public void Arrancar()
        {
            Console.WriteLine("El coche ha arrancado.");
        }

        public void Detener()
        {
            Console.WriteLine("El coche se ha detenido.");
        }

        static void Main(string[] args)
        {
            Metodos_SinParametros mp = new Metodos_SinParametros();
            mp.Arrancar();
            mp.Detener();

            Persona p = new Persona();
            p.Saludar();
            p.Dormir();

            Perro pe = new Perro();
            pe.Comer();
            pe.Ladrar();
        }
    }

    class Persona
    {
        public void Saludar()
        {
            Console.WriteLine("Hola, mucho gusto!");
        }

        public void Dormir()
        {
            Console.WriteLine("La persona está durmiendo...");
        }
    }

    class Perro
    {
        public void Ladrar()
        {
            Console.WriteLine("El perro está ladrando ¡Guau guau!");
        }

        public void Comer()
        {
            Console.WriteLine("El perro está comiendo su comida.");
        }
    }
}